src/main/java/com/faforever/client/fx/PlayerContextMenuController.java

```diff
@@ -0,0 +1,249 @@
+    JavaFxUtil.addAndTriggerListener(player.statusProperty(), weakChatUserPropertyListener);
+  }
+
+
+  public void onShowUserInfoSelected() {
+    PlayerBean player = getPlayer();
+    PlayerInfoWindowController playerInfoWindowController = uiService.loadFxml("theme/user_info_window.fxml");
+    playerInfoWindowController.setPlayer(player);
+    playerInfoWindowController.setOwnerWindow(getRoot().getOwnerWindow());
+    playerInfoWindowController.show();
+  }
+
+  public void onCopyUsernameSelected() {
+    PlayerBean player = getPlayer();
+    ClipboardUtil.copyToClipboard(player.getUsername());
+  }
+
+  public void onAddFriendSelected() {
+    PlayerBean player = getPlayer();
+    if (player.getSocialStatus() == FOE) {
+      playerService.removeFoe(player);
+    }
+    playerService.addFriend(player);
+  }
+
+  public void onRemoveFriendSelected() {
+    PlayerBean player = getPlayer();
+    playerService.removeFriend(player);
+  }
+
+  public void onAddFoeSelected() {
+    PlayerBean player = getPlayer();
+    if (player.getSocialStatus() == FRIEND) {
+      playerService.removeFriend(player);
+    }
+    playerService.addFoe(player);
+  }
+
+  public void onRemoveFoeSelected() {
+    PlayerBean player = getPlayer();
+    playerService.removeFoe(player);
+  }
+
+  public void onReport() {
+    ReportDialogController reportDialogController = uiService.loadFxml("theme/reporting/report_dialog.fxml");
+    PlayerBean player = getPlayer();
+    reportDialogController.setOffender(player);
+    reportDialogController.setOwnerWindow(getRoot().getOwnerWindow());
+    reportDialogController.show();
+  }
+
+  public void onJoinGameSelected() {
+    PlayerBean player = getPlayer();
+    joinGameHelper.join(player.getGame());
+  }
+
+  public void onWatchGameSelected() {
+    PlayerBean player = getPlayer();
+    try {
+      replayService.runLiveReplay(player.getGame().getId());
+    } catch (Exception e) {
+      log.error("Cannot display live replay", e.getCause());
+      notificationService.addImmediateErrorNotification(e, "replays.live.loadFailure.message");
+    }
+  }
+
+  public void onViewReplaysSelected() {
+    PlayerBean player = getPlayer();
+    eventBus.post(new ShowUserReplaysEvent(player.getId()));
+  }
+
+  public void onKickGame() {
+    moderatorService.closePlayersGame(getPlayer());
+  }
+
+  public void onKickLobby() {
+    moderatorService.closePlayersLobby(getPlayer());
+  }
+
+  protected void setModeratorOptions(Set<String> permissions, PlayerBean player) {
+    boolean notSelf = !player.getSocialStatus().equals(SELF);
+
+    JavaFxUtil.runLater(() -> {
+      kickGameItem.setVisible(notSelf & permissions.contains(GroupPermission.ADMIN_KICK_SERVER));
+      kickLobbyItem.setVisible(notSelf & permissions.contains(GroupPermission.ADMIN_KICK_SERVER));
+      broadcastMessage.setVisible(notSelf & permissions.contains(GroupPermission.ROLE_WRITE_MESSAGE));
```

Comment:
Why does broadcast message need to be notSelf?

Reply:
Apparently it does not. I'll also transfer it to only the chat context menu, because I think it makes the most sense there.